app/containers/Profile/index.js

```diff
@@ -8,29 +8,262 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
+import { Helmet } from 'react-helmet';
+import { translationMessages } from 'i18n';
 
 import injectSaga from 'utils/injectSaga';
 import injectReducer from 'utils/injectReducer';
-import makeSelectProfile from './selectors';
+import PropTypes from 'prop-types';
+
+import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
+import { makeSelectAccount } from 'containers/AccountInitializer/selectors';
+import { getBlob } from 'utils/profileManagement';
+
+import {
+  DISPLAY_NAME_FIELD,
+  POSITION_FIELD,
+  COMPANY_FIELD,
+  ABOUT_FIELD,
+} from './constants';
+
+import {
+  getProfileInfo,
+  uploadImageFileAction,
+  editImage,
+  clearImageChanges,
+  saveImageChanges,
+  saveProfileAction,
+  getCitiesList,
+  chooseLocation,
+} from './actions';
+
 import reducer from './reducer';
 import saga from './saga';
+import messages from './messages';
+import * as profileSelectors from './selectors';
+
+import ProfileForm from './ProfileForm';
+import Preloader from './Preloader';
+import NoSuchUser from './NoSuchUser';
+import Wrapper from './Wrapper';
 
 /* eslint-disable react/prefer-stateless-function */
 export class Profile extends React.Component {
+  componentWillMount() {
+    this.uploadImage = this.uploadImage.bind(this);
+    this.getCroppedAvatar = this.getCroppedAvatar.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    Profile.saveProfile = Profile.saveProfile.bind(this);
+    Profile.saveProfileActionDtch = this.props.saveProfileActionDtch.bind(this);
+  }
+
+  /*
+   * @componentDidMount: getting info about user profile
+   */
+
+  componentDidMount() {
+    const userKey = Profile.getUserKey(window.location.pathname);
```

Comment:
Use route parameters instead of manually parsing URL
https://tylermcginnis.com/react-router-url-parameters/